src/main/java/org/dice_group/grp/grammar/digram/DigramHelper.java

```diff
@@ -47,14 +68,275 @@
+		Map<Digram, Set<DigramOccurence>> nonOverlMap = new HashMap<Digram, Set<DigramOccurence>>();
+		Map<Digram, Set<DigramOccurence>> map = DigramHelper.mapDigrams(occurrences);
+		List<Digram> sortedDigrams = DigramHelper.sortDigrambyFrequence(map.keySet());
+		
+		Set<RDFNode> visitedNodes = new HashSet<RDFNode>();
+		
+		for(Digram curDigram: sortedDigrams) {
+			Set<DigramOccurence> digramOccurences = map.getOrDefault(curDigram, new HashSet<DigramOccurence>());
+			for(DigramOccurence curOccurrence: digramOccurences) {
+				Set <RDFNode> nodes = curOccurrence.getNodes();
+
+				// adds the occurrence if there's no nodes in common
+				if(Collections.disjoint(visitedNodes, nodes)) {
+					if(nonOverlMap.get(curDigram) != null)
+						nonOverlMap.get(curDigram).add(curOccurrence);
+					else {
+						Set<DigramOccurence> tempOccur = new HashSet<DigramOccurence>();
+						tempOccur.add(curOccurrence);
+						nonOverlMap.put(curDigram, tempOccur);
+					}
+				}
+				
+				// if empty, add the first occurrence anyways
+				if(nonOverlMap.isEmpty()) {
+					Set<DigramOccurence> tempOccur = new HashSet<DigramOccurence>();
+					tempOccur.add(curOccurrence);
+					nonOverlMap.put(curDigram, tempOccur);
+				}
+				visitedNodes.addAll(nodes);
+			}
+		}
+		updateDigramCount(nonOverlMap);
+		return nonOverlMap;
+	}
+	
+	/**
+	 * Checks which of the nodes of both statements, are connected to the overall graph
+	 * @param stmt1
+	 * @param stmt2
+	 * @param graph
+	 * @param digramCase 
+	 * @return the set of external nodes
+	 */
+	public static Set<RDFNode> findExternals(List<RDFNode> nodes, Statement stmt1, Statement stmt2, Model graph, String digramCase){
+		Set<RDFNode> externals = new HashSet<RDFNode>();
+		for(RDFNode node: nodes) {
+			int diff = 1;
+			if(node.equals(nodes.get(1))) {
+				diff = 2;
+			} 
+			boolean isNodeExternal = isNodeExternal(getNodeName(node), diff, graph);
+			
+			if(isNodeExternal)
+				externals.add(node);
+		}
+		
+		return externals;
+	}
+	
+	/**
+	 * Returns true if the triple count for a given node is positive
+	 * @param nodeString
+	 * @param diff
+	 * @param graph
+	 * @return
+	 */
+	private static boolean isNodeExternal(String nodeString, int diff, Model graph) {
+		StringBuilder query = new StringBuilder();
+		query.append("SELECT (COUNT(*)-").append(diff).append(" AS ?total) WHERE { { ")
+		.append(nodeString).append(" ?p ?n . } UNION { ?s ?p1 ").append(nodeString).append(" . } }");	
+		
+		QueryExecution queryExecution = queryModel(graph, query.toString());
+		List<QuerySolution> results = selectModel(queryExecution);
+		for(QuerySolution solution: results) {
+			long count = solution.getLiteral("total").getLong();
+			if(count > 0) {
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	/**
+	 * Returns a string encapsulated with < > in case of an URI resource and " " in case of a literal
+	 * @param node
+	 * @return
+	 */
+	private static String getNodeName(RDFNode node) {
```

Comment:
There is a class called JenaNodeFormatter.format() in the HDT library (included) this does exactly that :)